addons/libs/dialog.lua

```diff
@@ -0,0 +1,165 @@
+-- This library was written to help find the ID of a known
+-- action message corresponding to an entry in the dialog tables.
+-- While the IDs can be collected in-game, they occasionally
+-- change and would otherwise need to be manually updated.
+-- It can also be used to find and decode an entry given the ID.
+
+-- Common parameters:
+-- dat: The entire content of the zone dialog DAT file
+-- i.e. local dat = io.open('path/to/dialog/DAT', 'rb'):read('*a')
```

Comment:
An interesting overload might be a function that does not take the entire DAT as a string, but a file handle (or path, and get its own handle) and only read as much as needed while iterating. That way the entire DAT wouldn't need to be kept in memory. But just an idea for a possible future version of this file.

Reply:
That would be cool. Especially for `get_entry`.